### PR TITLE
support for nutrimatic queries, improved ANSWERIZE and TO_MORSE

### PR DESCRIPTION
**Pull request by ubragg**
Improvements include:
- added a second side panels for Nutrimatic queries similar to the existing Wordsmith one (and refactored code to reuse the same template)
- improved ANSWERIZE function to strip all non-alphanumeric characters, but added an optional boolean argument to get the old behavior
- added NUTRIMATIC function that works similarly to the existing ANAGRAM function, but for Nutrimatic queries
- fixes TO_MORSE to work on more than one character at a time
- updated JSDoc on all the functions to make their usage more clearly documented in autocomplete

### CODE REVIEW

File: sheet_functions/general.js [hunk 1]
```diff
@@ -55,12 +70,50 @@ function fetchAnagramsFromWordsmith(anagramText, maxResults) {
...
+ *
+ * @customfunction
+ */
+function NUTRIMATIC(input, maxResults) {
+  return [fetchResultsFromNutrimatic(input, maxResults)];
+}
+
+
+/**
+ * Gets anagrams from wordsmith.org/anagram.
```
Comment (derricks) [inline sheet_functions/general.js]: Update to say nutrimatic :)

File: sheet_functions/general.js [hunk 2]
```diff
@@ -55,12 +70,50 @@ function fetchAnagramsFromWordsmith(anagramText, maxResults) {
...
+}
+
 /**
  * Given an input string, split each character into its own cell.
  *
- * @param {String} text to split up
- * @param {String} character to split on (optional, defaults to empty space)
- * @return {Array} the individual characters of the input text.
+ * @param {string} input text to split up
+ * @param {string=} delimiter character to split on (default="")
```
Comment (derricks) [inline sheet_functions/general.js:114]: didn't know about this option. neat.

File: sheet_functions/general.js [hunk 3]
```diff
@@ -151,39 +209,44 @@ function TERNARY_TO_NUMBER(ternaryString) {
...
     // convert word to dots and dashes
-    if (optDotChar != undefined || optDotChar != null) {
-      normalizedString = _gsub(normalizedString, optDotChar, '.');
+    if (dotChar != undefined || dotChar != null) {
+      normalizedString = _gsub(normalizedString, dotChar, '.');
     }
 
-    if (optDashChar != undefined || optDashChar != null) {
-      normalizedString = _gsub(normalizedString, optDashChar, '-');
+    if (dashChar != undefined || dashChar != null) {
```
Comment (derricks) [inline sheet_functions/general.js]: Not crucial to fix, but I obviously added the _isNullOrUndefined check after writing this :)

**Review by derricks — COMMENTED** [review]
Looks great (suggested fixes are nits). I had definitely intended to refactor the UI stuff if I ever added another one, so thanks for doing that.

**Comment by ubragg** [comment]
OK, comments addressed, tests updated and verified.

FYI, I also changed the unknown character from "??" to "?".  This works better in particular for those functions that typically output only a single character, since that might then become the input for something else that assumes one character.

File: sheet_functions/general.js (hunk 1 above)
Comment (ubragg) [reply sheet_functions/general.js]: fixed

File: sheet_functions/general.js (hunk 3 above)
Comment (ubragg) [reply sheet_functions/general.js]: fixed, this actually had a bug before that was always evaluating the inside.  everything is either not-undefined or not-null!

**Pull request merged**